cognite/client/data_classes/data_modeling/data_types.py

```diff
@@ -68,44 +68,58 @@ def __load_unit_ref(data: dict) -> UnitReference | None:
-            return Boolean(is_list=resource["list"])
-        elif type_ == "float32":
-            return Float32(is_list=resource["list"], unit=cls.__load_unit_ref(resource))
-        elif type_ == "float64":
-            return Float64(is_list=resource["list"], unit=cls.__load_unit_ref(resource))
-        elif type_ == "int32":
-            return Int32(is_list=resource["list"], unit=cls.__load_unit_ref(resource))
-        elif type_ == "int64":
-            return Int64(is_list=resource["list"], unit=cls.__load_unit_ref(resource))
-        elif type_ == "timestamp":
-            return Timestamp(is_list=resource["list"])
-        elif type_ == "date":
-            return Date(is_list=resource["list"])
-        elif type_ == "json":
-            return Json(is_list=resource["list"])
-        elif type_ == "timeseries":
-            return TimeSeriesReference(is_list=resource["list"])
-        elif type_ == "file":
-            return FileReference(is_list=resource["list"])
-        elif type_ == "sequence":
-            return SequenceReference(is_list=resource["list"])
-        elif type_ == "direct":
-            return DirectRelation(
-                container=ContainerId.load(container) if (container := resource.get("container")) else None,
-                # The PropertyTypes are used as both read and write objects. The `list` was added later
-                # in the API for DirectRelations. Thus, we need to set the default value to False
-                # to avoid breaking changes. When used as a read object, the `list` will always be present.
-                is_list=resource.get("list", False),
-            )
-        elif type_ == "enum":
-            values = {key: EnumValue._load(value) for key, value in resource["values"].items()}
-            return Enum(values=values, unknown_value=resource.get("unknownValue"))
-        logger.warning(f"Unknown property type: {type_}")
-        return cast(Self, UnknownCogniteObject(resource))
+        match type_ := resource["type"]:
+            case "text":
+                return Text(
+                    is_list=resource["list"],
+                    max_list_size=resource.get("maxListSize"),
+                    collation=resource.get("collation", "ucs_basic"),
+                )
+            case "boolean":
+                return Boolean(
+                    is_list=resource["list"],
+                    max_list_size=resource.get("maxListSize"),
+                )
+            case "float32" | "float64" | "int32" | "int64":
+                cls_map: dict[str, type[PropertyTypeWithUnit]] = {
+                    "float32": Float32,
+                    "float64": Float64,
+                    "int32": Int32,
+                    "int64": Int64,
+                }
+                return cls_map[type_](
+                    is_list=resource["list"],
+                    max_list_size=resource.get("maxListSize"),
+                    unit=cls.__load_unit_ref(resource),
```

Comment:
(my comment here got lost) This wasn't being loaded correctly previously - we should have a unit test for it.

Reply:
nvm. it was being loaded 🤦

Reply:
a few more unit-unit tests doesn't hurt 😂